src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -1,195 +1,195 @@
-        logger.log("Processing Bucket: " + bucketName);
-
-        ObjectListing files = s3Client.listObjects(bucketName);
-        List<KeyVersion> filesProcessed = new ArrayList<DeleteObjectsRequest.KeyVersion>();
-
-        for (Iterator<?> iterator = files.getObjectSummaries().iterator(); iterator.hasNext(); ) {
-            S3ObjectSummary summary = (S3ObjectSummary) iterator.next();
-            logger.log("Reading Object: " + summary.getKey());
-
-            String trackingNumber = summary.getKey().split("--")[0];
-            Pair<Long, String> lastKnownStatus = latestStatusForTrackingNumber.get(trackingNumber);
-
-            // Check if this shipment has already been delivered, skip this file
-            if (lastKnownStatus != null && "DELIVERED".equals(lastKnownStatus.getRight())) {
-                continue;
-            }
-
-            String fileContents = s3Client.getObjectAsString(bucketName, summary.getKey());
-
-            if (!isValidFile(fileContents)) {
-                logger.log(String.format("Skipping invalid file %s", summary.getKey()));
-                continue;
+    
+        public ShopifyShop connectToShopify(String subdomain) {
+            final String token = "shpss_sdkfhkjh134134141341344133412312345678";
+            final ShopifySdk shopifySdk = ShopifySdk.newBuilder()
+                 .withSubdomain(subdomain)
+                 .withAccessToken(token).build();
+            return shopifySdk.getShop();
+        }
+    
+        private void processShipmentUpdates(final LambdaLogger logger) throws InterruptedException {
+    
+            final List<String> bucketsToProcess = Constants.BUCKETS_TO_PROCESS;
+            final Map<String, Pair<Long, String>> latestStatusForTrackingNumber = new HashMap<String, Pair<Long, String>>();
+            final Map<String, List<KeyVersion>> filesToDelete = new HashMap<String, List<DeleteObjectsRequest.KeyVersion>>(); 
+            for (final String bucketName : bucketsToProcess) {
+                final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber);
+                filesToDelete.put(bucketName, filesProcessed);
             }
+            final AmazonS3 s3Client = EventHandler.getS3Client();
+    
+            //Create a new file in the Constants.SUMMARY_BUCKET
+            logger.log("Map of statuses -> " + latestStatusForTrackingNumber);
+            String summaryUpdateName = Long.toString(System.currentTimeMillis());
             
-            if (!fileContents.contains("\n")) {
-                
+            EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+            
+            long expirationTime = System.currentTimeMillis() + Duration.ofMinutes(1).toMillis();
+            while(System.currentTimeMillis() < expirationTime) {
+                if (s3Client.doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code appears to be waiting for a resource before it runs. You could use the waiters feature to help improve efficiency. Consider using ObjectExists or ObjectNotExists. For more information, see [https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/](https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/)